src/nnet3/nnet-simple-component.cc

```diff
@@ -119,15 +119,21 @@ void DropoutComponent::Propagate(const ComponentPrecomputedIndexes *indexes,
 
   BaseFloat dropout = dropout_proportion_;
   KALDI_ASSERT(dropout >= 0.0 && dropout <= 1.0);
-
+  // randomize the dropout matrix by row,
+  // i.e. [[1,1,1,1],[0,0,0,0],[0,0,0,0],[1,1,1,1],[0,0,0,0]]
   // This const_cast is only safe assuming you don't attempt
   // to use multi-threaded code with the GPU.
   const_cast<CuRand<BaseFloat>&>(random_generator_).RandUniform(out);
-
-  out->Add(-dropout); // now, a proportion "dropout" will be <0.0
-  out->ApplyHeaviside(); // apply the function (x>0?1:0).  Now, a proportion "dropout" will
-                         // be zero and (1 - dropout) will be 1.0.
-
+  // now, a proportion "dropout" will be <0.0
```

Comment:
There doesn't necessarily have to be an option in the scripts, it can be hardcoded; and anyway we'll consider that later; but the component should support both options.  e.g. adding "dropout-per-frame=true" in the config file to the component, will set the per-frame mode.